packages/google_maps_flutter/lib/src/google_maps_controller.dart

```diff
@@ -27,66 +27,65 @@ class GoogleMapController {
     return _controller.future;
   }
 
-  /// The Flutter widget that will contain the rendered Map. Used for caching.
-  Widget? get widget {
-    if (_widget == null && !_streamController.isClosed) {
-      _widget = WebView(
-        javascriptMode: JavascriptMode.unrestricted,
-        onWebViewCreated: (WebViewController webViewController) async {
-          _controller.complete(webViewController);
-
-          final String options = _createOptions();
-          final String initMap = '''
+  Future<void> _loadHtmlFromAssets() async {
+    final String options = _createOptions();
+    final String initMap = '''
           function initMap() {
             map = new google.maps.Map(document.getElementById('map'), $options);
             map.addListener('bounds_changed', BoundChanged.postMessage);
+            map.addListener('idle', Idle.postMessage);
+            map.addListener('click', (event) => Click.postMessage(JSON.stringify(event)));
+            map.addListener('rightclick', (event) => RightClick.postMessage(JSON.stringify(event)));
           }
           ''';
 
-          String html = await rootBundle.loadString('assets/map.html');
-          final int pos = html.indexOf('let map;');
-          html = html.substring(0, pos + 8) +
-              initMap +
-              html.substring(pos + 8, html.length);
-
-          final WebViewController mapView = await controller;
-          mapView.loadUrl(Uri.dataFromString(html,
-                  mimeType: 'text/html', encoding: Encoding.getByName('utf-8'))
-              .toString());
-        },
-        onProgress: (int progress) {
-          print('Google map is loading (progress : $progress%)');
-        },
-        javascriptChannels: <JavascriptChannel>{
-          _javascriptEventChannel(),
-        },
-        onPageStarted: (String url) {
-          print('Google map started loading.');
-        },
-        onPageFinished: (String url) async {
-          print('Google map finished loading.');
-        },
-        gestureNavigationEnabled: true,
-      );
-    }
-    return _widget;
+    String html = await rootBundle.loadString('assets/map.html');
```

Comment:
Currently, map.html is owned by example. IMO, I think that the plugin implementation can have it directly as a string. It seems that you will have to think about how to get the API key from the outside of this plugin

Reply:
Yes. Even though I have no any good idea yet, we have to consider to solve this issue.

Reply:
All right, let's find a way to solve it later.